tests/test_resource_warning.py

```diff
@@ -0,0 +1,81 @@
+import asyncio
+import time
+import typing
+import warnings
+from datetime import datetime
+
+import pytest
+
+import aiohttp
+
+last_warning: typing.Optional[str] = None
+
+
+def print_warnings(message, category, filename, lineno, file=None, line=None):
+    global last_warning
+    text = f'{datetime.now()} | ' \
+           f'{filename}:{lineno} {category.__name__:s}:{message}'
+    print(text)
+    # These are warnings in the 4.0a and prevent execution of the test
+    if "RuntimeWarning:coroutine 'noop' was never awaited" in text:
+        return None
+    last_warning = text
+
+
+@pytest.fixture()
+def warning_setup():
+    # setup warning
+    old_func = warnings.showwarning
+    warnings.showwarning = print_warnings
+    warnings.simplefilter('default')
+
+    yield
+
+    warnings.showwarning = old_func
```

Comment:
This can be simplified as
```suggestion
def warning_setup(monkeypatch):
    monkeypatch.setattr(warnings, 'showwarning', print_warnings)
    warnings.simplefilter('default')
```